packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -407,6 +407,10 @@ export default util.createRule({
             propertyText = getMemberExpressionText(node.property);
             break;
 
+          case AST_NODE_TYPES.BinaryExpression:
+            propertyText = sourceCode.getText(node.property);
+            break;
```

Comment:
There's already a `case` a few lines above for `AST_NODE_TYPES.Literal` and `AST_NODE_TYPES.TemplateLiteral`. Could you add this there please?